sdk/opendp/smartnoise/synthesizers/quail.py

```diff
@@ -0,0 +1,108 @@
+import numpy as np
+import pandas as pd
+import pickle
+
+from opendp.whitenoise.synthesizers.base import SDGYMBaseSynthesizer
+
+class QUAILSynthesizer(SDGYMBaseSynthesizer):
+    """
+    Quailified Architecture to Improve Labeling.
+
+    Divide epsilon in a known classification task
+    between a differentially private synthesizer and
+    classifier. Train DP classifier on real, fit DP synthesizer
+    to features (excluding the target label)
+    and use synthetic data from the DP synthesizer with
+    the DP classifier to create artificial labels. Produces
+    complete synthetic data.
+
+    More information here: 
+    Differentially Private Synthetic Data: Applied Evaluations and Enhancements
+    https://arxiv.org/abs/2011.05537
+    """
+    
+    def __init__(self, epsilon, dp_synthesizer, synth_args, dp_classifier, class_args, target, test_size=0.2, seed=42, eps_split=0.9):

```

Comment:
It might be a simpler API to have

(epsilon,  lambda epsilon: returns synthesizer, lambda epsilon: return classifier, target, seed, eps_split)

I do not think we need the train test split done internally within QUAIL

Reply:
And func that returns the object is more flexible on inputs as well as epsilon handling.

For example, how would key word arguments work with synth_args?

Reply:
The public API should include
`

@property 
def classifier/model

@property 
def synthesizer`

Also a test please :)

I am so excited to ship this feature. Great idea and thanks for the contribution!

The run time to utility tradeoff is one of the more exciting aspects of the approach